java-squid/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -104,7 +104,12 @@ public boolean isSubtype(JavaType t, JavaType s) {
   public Type leastUpperBound(List<Type> types) {
     Preconditions.checkArgument(types.size() > 1);
 
-    List<Set<Type>> supertypes = supertypes(types);
+    List<Set<Type>> supertypes;
+    try {
+      supertypes = supertypes(types);
+    } catch (GenericClassDetectedException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)